app/src/org/commcare/views/media/CustomVideoView.java

```diff
@@ -0,0 +1,49 @@
+package org.commcare.views.media;
+
+import android.content.Context;
+import android.widget.VideoView;
+import java.util.Date;
+
+/**
+ * A custom VideoView which records the total video play duration
+ * and send the update via {@link org.commcare.views.media.CustomVideoView.TimeSpentListener#getPlayDuration(long) getPlayDuration}
+ * @author $|-|!˅@M
+ */
+public class CustomVideoView extends VideoView {
+
+    private TimeSpentListener listener;
+
+    private long duration;
+    private long startTime;
+
+    public CustomVideoView(Context context) {
+        super(context);
+    }
+
+    public void setListener(TimeSpentListener listener) {
+        this.listener = listener;
+    }
+
+    @Override
+    public void pause() {
+        super.pause();
+        duration += new Date().getTime() - startTime;
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        startTime = new Date().getTime();
```

Comment:
what if user clicks Play -> Pause -> Play again ?  this won't count the time already elapsed in the first play to pause lap.

Reply:
We'll store the total time played so far in duration variable here https://github.com/dimagi/commcare-android/pull/2166/files/c29131841b5431d2bb628075bb7fea6f7a454a87#diff-1a0cb3908dcc92f87e97b0f9d670d968R30

Reply:
Ahh cool. Thanks for clarifying.